citadel/spawn_urdf.md

```diff
@@ -0,0 +1,56 @@
+# Spawn URDF
+
+This tutorial will cover how to spawn a URDF model in Ignition Gazebo.
+URDF files are often used in ROS to represent robot models.
+While SDF can describe a world with multiple robot models, URDF can only describe one robot model.
+More information about URDF can be found at https://wiki.ros.org/urdf.
+
+## Obtaining a URDF file
+
+This tutorial assumes that the reader already has a URDF file that they'd like to load into Ignition Gazebo.
+If you are looking for a URDF file to use, you can use the `rrbot` robot model in the [`gazebo_ros_demos`](https://github.com/ros-simulation/gazebo_ros_demos) package, or use one of the examples listed at https://wiki.ros.org/urdf/Examples.
+
+If you have a `xacro` representation of a robot model, you can turn the `xacro` file into a `URDF` file using the [`xacro`](https://index.ros.org/p/xacro/) package: see [this tutorial](https://docs.ros.org/en/foxy/Tutorials/URDF/Using-Xacro-to-Clean-Up-a-URDF-File.html) for more information.
+
+## Spawning the URDF
+
+To spawn a URDF model in Ignition Gazebo, we will start a world and make use of the world's `create` service, which uses the [EntityFactory](https://ignitionrobotics.org/api/msgs/8.1/classignition_1_1msgs_1_1EntityFactory.html) message type.
+
+Start by launching an empty world in Ignition Gazebo:
+```
+ign gazebo empty.sdf
+```
+
+You should see a window that looks like this:
+
+![empty_world](tutorials/spawn_urdf/empty_world.png)
+
+In another terminal, get the list of available services by running `ign service -l`.
+Look for a `create` service.
+You should see this service in the list:
+```
+/world/empty/create
+```
+
+We can double-check that this is the service we want to use by running `ign service -is /world/empty/create`.
+This will show us the service's request and response message types:
+```
+ignition.msgs.EntityFactory, ignition.msgs.Boolean
+```
+
+Now that we have found the service that has an `EntityFactory` request type, we can call this service and pass the URDF file to the service's request so that the robot represented by this URDF file is spawned in the Ignition Gazebo world.
+We do this by setting the desired URDF file to the `sdf_file_name` field of the `EntityFactory` request message.
+[SDFormat](https://ignitionrobotics.org/libs/sdformat) will then internally convert the URDF file to an SDF representation, and load this into the running world.
```

Comment:
we generally use `SDFormat` to refer to the specification and `libsdformat` to refer to the parser library. So perhaps: "The [libsdformat](...) library will then internally convert..."